metricproviders/datadog/datadog.go

```diff
@@ -0,0 +1,172 @@
+			return metricutil.MarkMeasurementError(newMeasurement, err)
+		}
+
+		if len(response) < 1 {
+			time.Sleep(1 * time.Second)
+		}
+	}
+
+	if len(response) > 0 {
+		newValue, newStatus, err := p.processResponse(metric, response[0])
+		if err != nil {
+			return metricutil.MarkMeasurementError(newMeasurement, err)
+		}
+		newMeasurement.Value = newValue
+		newMeasurement.Phase = newStatus
+		finishedTime := metav1.Now()
+		newMeasurement.FinishedAt = &finishedTime
+		return newMeasurement
+	}
+	newMeasurement.Value = ""
+	newMeasurement.Phase = v1alpha1.AnalysisPhaseInconclusive
+	finishedTime := metav1.Now()
+	newMeasurement.FinishedAt = &finishedTime
+	return newMeasurement
+}
+
+// Resume should not be used with the datadog provider since all the work should occur in the Run method
+func (p *Provider) Resume(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric, measurement v1alpha1.Measurement) v1alpha1.Measurement {
+	p.logCtx.Warn("Datadog provider should not execute the Resume method")
+	return measurement
+}
+
+// Terminate should not be used with the datadog provider since all the work should occur in the Run method
+func (p *Provider) Terminate(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric, measurement v1alpha1.Measurement) v1alpha1.Measurement {
+	p.logCtx.Warn("Datadog provider should not execute the Terminate method")
+	return measurement
+}
+
+// GarbageCollect is a no-op for the datadog provider
+func (p *Provider) GarbageCollect(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric, limit int) error {
+	return nil
+}
+
+func (p *Provider) processResponse(metric v1alpha1.Metric, response dd.Series) (string, v1alpha1.AnalysisPhase, error) {
+	length := len(response.Points)
+	switch {
+	case length < 1:
+		return "", v1alpha1.AnalysisPhaseInconclusive, nil
+	// If we get 1 or more points, select the first point
+	case length >= 1:
+		result := *response.Points[0][1]
+		valueStr := fmt.Sprintf("%f", result)
+		newStatus := evaluate.EvaluateResult(result, metric, p.logCtx)
+		return valueStr, newStatus, nil
+	// TODO (cabrinha) add other response types
+	default:
+		return "", v1alpha1.AnalysisPhaseError, fmt.Errorf("Datadog metric type not supported")
+	}
+}
+
+// NewDatadogProvider creates a new Datadog client
+func NewDatadogProvider(client dd.Client, logCtx log.Entry) *Provider {
+	return &Provider{
+		logCtx: logCtx,
+		client: client,
+	}
+}
+
+// NewDatadogAPI generates a datadog API from the metric configuration
+func NewDatadogAPI(metric v1alpha1.Metric, kubeclientset kubernetes.Interface) (*dd.Client, error) {
+	ns := Namespace()
+	secret, err := kubeclientset.CoreV1().Secrets(ns).Get(DatadogSecretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(secret.Data[`datadog-api-key`]) > 0 && len(secret.Data[`datadog-app-key`]) > 0 {
+		apiKey := fmt.Sprintf("%s", secret.Data[`datadog-api-key`])
+		appKey := fmt.Sprintf("%s", secret.Data[`datadog-app-key`])
+		client := dd.NewClient(apiKey, appKey)
```

Comment:
I think this approach only allows you to have one baseUrl since you can't specify a datadog-api-key and datadog-app-key for each endpoint.

Reply:
Yeah, I think that's what we want though? Just one base url, just one client.